Environment_Engine/Query/InvalidPanels.cs

```diff
@@ -0,0 +1,144 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Environment;
+using BH.Engine.Geometry;
+using BH.oM.Environment.Elements;
+
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Environment
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Return a collection of panels which we identified Invalid data 1-external adj space, 2-shade check, 3-internal adj space, 4-internal two diff spaces, 5-area to small, 6-segment to short, 7-polyline closed, 8-poltyline interesect")]
+        [Input("panels", "A collection of environment panels")]
+        [Input("panelArea", "A min panel area to return warning")]
+        [Input("panelEdgeLength", "A min panelEdgeLength to return warning")]
+        [Output("invalidPanels", "A collection of panels which are invalid group in sublist for each of 8 checks that we perform, see description")]
+        public static List<List<Panel>> InvalidPanels(this List<Panel> panels, double panelArea = 0.15, double panelEdgeLength = 0.15)
+        {
+            List<List<Panel>> rtn = new List<List<Panel>>();
+
+            for (int i = 0; i < 8; i++)
+                rtn.Add(new List<Panel>());
+            foreach (Panel p in panels)
+
+            {
+                //check for external
```

Comment:
What does this comment serve to do? Comments in code should serve to explain what the code is doing that isn't obvious from other documentation around. This particular comment does not meet that standard. Either remove it, or make it more useful for describing what's going on.